### PR TITLE
Upgrade to finagle 6.39.0

### PR DESCRIPTION
**Pull request by adleong**

### CODE REVIEW

File: admin/src/main/scala/io/buoyant/admin/names/DelegateApiHandler.scala [hunk 1]
```diff
@@ -174,7 +174,7 @@ object DelegateApiHandler {
     private[this] val mapper = new ObjectMapper with ScalaObjectMapper
     mapper.registerModule(DefaultScalaModule)
     mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
-    mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL)
```
Comment (olix0r) [inline admin/src/main/scala/io/buoyant/admin/names/DelegateApiHandler.scala]: Is this related to the finagle change? Do we need to change our jackson version to be consistent with finagle?

File: router/core/src/test/scala/io/buoyant/router/RouterTest.scala [hunk 2]
```diff
@@ -120,7 +120,6 @@ class RouterTest extends FunSuite {
         val id: Any = path match {
           case Path.Utf8("0") => Path.Utf8("some", "path")
           case Path.Utf8("1") => "string"
-          case Path.Utf8("2") => null
```
Comment (olix0r) [inline router/core/src/test/scala/io/buoyant/router/RouterTest.scala]: What changed in finagle related to this?

Comment (adleong) [inline router/core/src/test/scala/io/buoyant/router/RouterTest.scala]: finagle now calls `.hashCode` on ids which means that using `null` for an id is a no-no.

File: admin/src/main/scala/io/buoyant/admin/names/DelegateApiHandler.scala (hunk 1 above)
Comment (adleong) [inline admin/src/main/scala/io/buoyant/admin/names/DelegateApiHandler.scala]: Yep.  Now updated to 2.8.4.
